packages/test/test-end-to-end-tests/src/test/signalPerfTelemetry.spec.ts

```diff
@@ -0,0 +1,367 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { describeCompat, itExpects } from "@fluid-private/test-version-utils";
+import { IContainer } from "@fluidframework/container-definitions/internal";
+import { ConnectionState } from "@fluidframework/container-loader";
+import { MockLogger } from "@fluidframework/telemetry-utils/internal";
+import {
+	DataObjectFactoryType,
+	ITestContainerConfig,
+	ITestFluidObject,
+	ITestObjectProvider,
+	getContainerEntryPointBackCompat,
+	timeoutPromise,
+} from "@fluidframework/test-utils/internal";
+
+const waitForSignal = async (...signallers: { once(e: "signal", l: () => void): void }[]) =>
+	Promise.all(
+		signallers.map(async (signaller, index) =>
+			timeoutPromise((resolve) => signaller.once("signal", () => resolve()), {
+				durationMs: 2000,
+				errorMsg: `Signaller[${index}] Timeout`,
+			}),
+		),
+	);
+
+describeCompat("Signal performance telemetry", "NoCompat", (getTestObjectProvider, apis) => {
+	let provider: ITestObjectProvider;
+	let dataObject: ITestFluidObject;
+	let container: IContainer;
+	let containerRuntime;
+	let logger: MockLogger;
+
+	beforeEach("getTestObjectProvider", async () => {
+		provider = getTestObjectProvider();
+
+		logger = new MockLogger();
+
+		const testContainerConfig: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			loaderProps: {
+				logger,
+			},
+		};
+		container = await provider.makeTestContainer(testContainerConfig);
+		dataObject = await getContainerEntryPointBackCompat<ITestFluidObject>(container);
+		containerRuntime = dataObject.context.containerRuntime;
+
+		// need to be connected to send signals
+		if (container.connectionState !== ConnectionState.Connected) {
+			await new Promise((resolve) => container.once("connected", resolve));
+		}
+	});
+
+	describe("for all drivers", () => {
+		itExpects(
+			"SignalLatency telemetry after 100 broadcast signals",
+			[
+				{
+					eventName: "fluid:telemetry:ContainerRuntime:SignalLatency",
+					clientType: "interactive",
+				},
+			],
+			async () => {
+				// Send 100+ broadcast signals to trigger latency telemetry event
+				for (let i = 0; i < 101; i++) {
+					containerRuntime.submitSignal("signal", "test");
+					await waitForSignal(containerRuntime);
```

Comment:
Seems "slow" and inefficient. Also, would expect linter to complain about await in a loop.
I know the help is for exactly one signal. If we want to be testing single round trip one at a time, then explain why in comments.
If not, then I would suggest regular listener registration that resolves once N signals have been received.

Reply:
Right -- updated waitForSignals to resolve once N signals have been received.